### DISCUSSION TITLE
lua调用go编写的so失败

### DISCUSSION THREAD

**Discussion by judithzzh** [cloudwu/skynet#1031] [Q&A]
有一个go写的so库，我用系统上装的lua5.3调用能正确执行，但是在skynet里就提示找不到函数
![image](https://user-images.githubusercontent.com/19266616/59321656-78a5de80-8d05-11e9-93c4-95ae8e026563.png)
上面是我打印出的这个lib
![image](https://user-images.githubusercontent.com/19266616/59321684-8ce9db80-8d05-11e9-94c7-abed15b4c989.png)
type时却是nil

**Comment by judithzzh** [comment]
云大能指点下吗

**Reply by BreezeCode** [reply]
楼主问题解决了么，能不能参考一下你的实现方式，我也想用go扩展一些不好实现的功能，但是不知如何下手

**Reply by hanxi** [reply]
这有个用 go 写服务的例子： https://github.com/xjdrew/skynet_go_service

**Comment by cloudwu** [comment]
1. 改掉截图的坏毛病吧，明明都是文本，学学 markdown 。
2. 多半是 lua 库的链接问题。

**Comment by judithzzh** [comment]
好的  我再查查。markdown不太熟，还在持续学习中，多谢~~
